src/ui/MLFacetManager/MLFacetManager.ts

```diff
@@ -0,0 +1,183 @@
+import { Component } from '../Base/Component';
+import { MLFacet } from '../MLFacet/MLFacet';
+import { InitializationEvents } from '../../events/InitializationEvents';
+import { QueryEvents, IQuerySuccessEventArgs } from '../../events/QueryEvents';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { exportGlobally } from '../../GlobalExports';
+import { find } from 'underscore';
+import { IFacetResponse } from '../../rest/Facet/FacetResponse';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { ComponentOptions } from '../Base/ComponentOptions';
+
+export interface IMLFacetManagerOptions {
+  enableReorder?: boolean;
+  onUpdate?: IMLFacetManagerOnUpdate;
+  compareFacets?: IMLFacetManagerCompareFacet;
+}
+
+export interface IMLFacetManagerOnUpdate {
+  (facet: MLFacet, index: number): void;
+}
+
+export interface IMLFacetManagerCompareFacet {
+  (facetA: MLFacet, facetB: MLFacet): number;
+}
+
+/**
+ * The `MLFacetManager` component is meant to be a parent for multiple [MLFacet]{@link MLFacet} components.
+ * It allows more control over the rendering and ordering of the children [MLFacet]{@link MLFacet} components.
+ */
+export class MLFacetManager extends Component {
+  static ID = 'MLFacetManager';
+  static doExport = () => exportGlobally({ MLFacetManager });
+
+  /**
+   * The options for the MLFacet
+   * @componentOptions
+   */
+  static options: IMLFacetManagerOptions = {
+    /**
+     * Whether to allow the reordering of facets based on Coveo ML and index ranking scores.
+     *
+     * **Default:** `true`
+     */
+    enableReorder: ComponentOptions.buildBooleanOption({ defaultValue: true, section: 'Filtering' }),
+    /**
+     * The event handler function to execute when facets are updated in the query response.
+     * For each `MLFacet`, you can provide a callback function to execute after the facet is reordered.
+     *
+     * **Note:**
+     * > You cannot set this option directly in the component markup as an HTML attribute. You must either set it in the
+     * > [`init`]{@link init} call of your search interface (see
+     * > [Components - Passing Component Options in the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsintheinitCall)),
+     * > or before the `init` call, using the `options` top-level function (see
+     * > [Components - Passing Component Options Before the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsBeforetheinitCall)).
+     */
+    onUpdate: ComponentOptions.buildCustomOption<IMLFacetManagerOnUpdate>(() => {
+      return null;
+    }),
+    /**
+     * A function that defines the sort order for the facets.
+     *
+     * Called on every successful query reponse.
+     *
+     * The implementation of this method is identical to the Javascript `compareFunction` of the sort method for arrays.
```

Comment:
🤔 I'm not sure what that means.

Is it that the default value for this option if you don't pass any is the default "compareFUnction", or that if you wanna pass one, you need to implement it similar to the "compareFunction", where you must return -1, 0, 1 ?

Also missing "what" you can do with with, and "why" you would use it, instead of just "how".

Reply:
> Here you can define a method to be executed when facets are being ordered after every successful query response. This hook can be used to implement custom ordering.
@olamothe @fbeaudoincoveo I had also provided an example that could be added in an article.

@olamothe yes if you pass a method you must return -1, 0, 1, exactly like the "compareFunction"

Reply:
Suggestion:

> A custom sort function to execute on facets on every successful query response (see [Implementing a Custom Dynamic Facet Sort Function](https://docs.coveo.com/en/2902/#implementing-a-custom-dynamic-facet-sort-function)).
>
> **Notes:**
> - If specified, the function must implement the JavaScript `compareFunction` (see [Array.prototype.sort(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)].
> - You cannot set this option directly (...)

Reply:
The link I'm providing points to the same article as the one for custom facet behaviors (just a different anchor in the page).

I'm thinking of naming the page "Customizing Dynamic Facets", and just putting everything there.